src/nvim/tui/tui.c

```diff
@@ -568,13 +568,15 @@ static void cursor_goto(UI *ui, int row, int col)
     ugrid_goto(grid, row, col);
     return;
   }
-  if (0 == col ? col != grid->col :
+  if (!(1 == row && 0 == col)
```

Comment:
Changing that to:

    if (!((1 == row || 1 == grid->row) && 0 == col)

seems to work better.